projects/core/src/checkout/store/effects/checkout.effect.ts

```diff
@@ -414,10 +415,52 @@ export class CheckoutEffects {
     withdrawOn(this.contextChange$)
   );
 
+  @Effect()
+  setCostCenter$: Observable<
+    | CheckoutActions.SetCostCenterSuccess
+    | CheckoutActions.SetCostCenterFail
+    | CheckoutActions.ClearCheckoutDeliveryMode
+    | CheckoutActions.ClearCheckoutDeliveryAddress
+    | CartActions.LoadCart
+  > = this.actions$.pipe(
+    ofType(CheckoutActions.SET_COST_CENTER),
+    map((action: any) => action.payload),
+    mergeMap((payload) => {
+      return this.checkoutCostCenterConnector
+        .setCostCenter(payload.userId, payload.cartId, payload.costCenterId)
+        .pipe(
+          mergeMap(() => [
```

Comment:
Perhaps a switchMap or concatMap?  We probably don't want multiple cost center updates happening in parallel.

Reply:
use `switchMap` in line 428